src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,151 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final int DEFAULT_SIZE = 0;
+    private static final int HALF_DIVIDER = 2;
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
+    private static class Node<E> {
+        private E item;
+        private Node<E> next;
+        private Node<E> prev;
+
+        Node(Node<E> prev, E element, Node<E> next) {
+            this.item = element;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
+    private Node<T> getNodeByIndex(int index) {
+        checkBoundsExclusive(index);
+        Node<T> currentNode;
+        if (index < size / HALF_DIVIDER) {
+            currentNode = head;
+            while (index-- > DEFAULT_SIZE) {
+                currentNode = currentNode.next;
+            }
+        } else {
+            currentNode = tail;
+            while (++index < size) {
+                currentNode = currentNode.prev;
```

Comment:
here we can use `2` instead of `HALF_DIVIDER`, it would be fine
same about `DEFAULT_SIZE`
also `(index-- > DEFAULT_SIZE)` looks cool but isn't so readable, 
let's compare and decrement `index` at different lines